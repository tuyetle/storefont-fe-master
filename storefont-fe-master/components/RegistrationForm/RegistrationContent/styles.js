import { css } from 'glamor';
import themes from '~/styles/themes';
import variables from '~/styles/variables';

export default {
  form: css({
    paddingTop: '3.313em',
    paddingBottom: '5.688em',
    margin: '0 auto',
    width: '80%',
    '@media (min-width: 768px)': {
      width: '70%',
    },
  }),
  title: css({
    ...themes['font-notoserif'],
    fontSize: '1.563em',
    textAlign: 'center',
  }),
  description: css({
    marginTop: '1em',
    textAlign: 'center',
    marginBottom: '2em',
  }),

  rulesAndCondition: css({
    textAlign: 'center',
    fontSize: '0.813em',
    '@media (min-width: 768px)': {
      margin: '0 4em',
    },
  }),
  linkRules: css({
    color: 'red',
  }),
  captcha: css({
    display: 'block',
    alignSelf: 'center',
    '& #g-recaptcha': {
      transform: 'scale(0.77)',
      '@media (min-width: 768px)': {
        transform: 'scale(1)',
        '& div': {
          margin: '0 auto',
        },
      },
    },
  }),
  checkbox: css({
    margin: '1.374em 0 1.875em 0',
    '& input[type=checkbox]': {
      verticalAlign: 'middle',
      marginRight: '0.688em',
    },
  }),

  input: css({
    marginTop: '0.875em',
    marginBottom: '0.188em',
    borderRadius: '0',
    borderBottom: '1px solid lightgray',
    border: '1px solid lightgray',
    '&:focus': {
      borderBottomColor: variables.$focusBorderBlue,
      boxShadow: 'none',
    },

    '&.form-control:focus': {
      boxShadow: 'none',
      border: '1px solid lightgray',
    },
  }),

  error: css({
    border: 'none',
    borderRadius: '0',
    borderBottom: '1px solid red',
  }),

  errorMsg: css({
    textAlign: 'center',
    color: 'red',
    fontSize: '0.8em',
    marginBottom: '1em',
  }),

  successMsg: css({
    marginBottom: '1em',
    textAlign: 'center',
    fontSize: '1em',
    color: variables.$colorSuccessMessage,
  }),

  submitButton: css({
    ...themes['submit-button'],
  }),

  disableSubmitButton: css({
    ...themes['disable-submit-button'],
  }),
};
