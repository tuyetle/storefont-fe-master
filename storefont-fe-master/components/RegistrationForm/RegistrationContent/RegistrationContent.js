import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { noop, isEmpty, isNil } from 'lodash';
import { Field } from 'react-form';
import Link from 'next/link';
import { translate } from 'react-i18next';
import EmailOrPhoneFormField from '~/components/common/Forms/EmailOrPhoneFormField/EmailOrPhoneFormField';
import RecaptchaFormField from '~/components/common/Forms/RecaptchaFormField/RecaptchaFormField';
import CheckboxFormField from '~/components/common/Forms/CheckboxFormField/CheckboxFormField';
import { RULES_AND_CONDITION } from '~/config/routerConfig';
import { RECAPTCHA_KEY } from '~/config/config';
import styles from './styles';

export class RegistrationContent extends PureComponent {
  validateCaptcha = value => (isEmpty(value) ? ({ error: 'captcha' }) : '');

  renderRegistrationForm = () => {
    const { t, registrationRequestResult, formApi } = this.props;

    if (!isNil(registrationRequestResult)) return null;

    return [
      <div key="registrationEmailOrPhone" className="form-group">
        <strong className={styles.label}>{t('emailOrPhone')}</strong>
        <Field
          field="emailOrPhone"
          component={EmailOrPhoneFormField}
          className={`${styles.input} form-control`}
        />
      </div>,
      <div key="registrationCaptcha" className={`${styles.captcha}`} >
        <Field
          field="captcha"
          component={RecaptchaFormField}
          sitekey={RECAPTCHA_KEY}
          hl="vi"
          isRequired
          validate={this.validateCaptcha}
        />
      </div>,
      <div key="registrationReceiveNews">
        <label className={`${styles.checkbox}`} >
          <Field
            component={CheckboxFormField}
            field="receiveNews"
          />
          <span >{t('receiveNews')}</span>
        </label>
      </div>,
      <input
        key="registrationSubmitAction"
        className={isEmpty(formApi.values) ? `${styles.disableSubmitButton}` : `${styles.submitButton}  btn rounded-0`}
        type="submit"
        disabled={isEmpty(formApi.values)}
        value={t('register')}
      />,
      <div key="registrationRulesAndConditions" className={`${styles.rulesAndCondition}`}>
        {t('rulesAndConditionBefore')}<Link href={RULES_AND_CONDITION} ><a className={`${styles.linkRules}`} href="/">{t('rulesAndCondition')}</a></Link>{t('rulesAndConditionAfter')}
      </div>,
    ];
  }

  render = () => {
    const { t, registrationRequestResult, formApi } = this.props;

    return (
      <form className={`${styles.form}`} onSubmit={formApi.submitForm}>
        <div className={`${styles.title}`} > {t('register')} </div>
        <div className={`${styles.description}`} > {t('registerDescription')}</div>
        <div className={`${styles.successMsg}`}>
          {registrationRequestResult === true && <span>{t('successRegistrationMsg')}</span>}
        </div>
        <div className={`${styles.errorMsg}`}>
          {registrationRequestResult === false && <span>{t('errorRegistrationMsg')}</span>}
        </div>
        {this.renderRegistrationForm()}
      </form>
    );
  }
}

RegistrationContent.propTypes = {
  formApi: PropTypes.shape({
    submitForm: PropTypes.func,
  }).isRequired,
  t: PropTypes.func,
  registrationRequestResult: PropTypes.bool,
};

RegistrationContent.defaultProps = {
  t: noop,
  registrationRequestResult: null,
};

export default translate(['auth'])(RegistrationContent);
