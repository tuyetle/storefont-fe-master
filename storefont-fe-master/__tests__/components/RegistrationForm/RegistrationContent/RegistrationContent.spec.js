import React from 'react';
import RegistrationContent from '~/components/RegistrationForm/RegistrationContent/RegistrationContent';
import renderer from 'react-test-renderer';

describe('(Component) RegistrationContent', () => {
  let component;
  let props;
  let spyCallback;

  beforeEach(() => {
    props = {
      t: value => value,
      formApi: { submitForm() {} },
      status: null,
    };
    spyCallback = jest.spyOn(props.formApi, 'submitForm');
    component = mount(<RegistrationContent {...props} />);
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call submitForm when click submit', () => {
    const btn = component.find('input[type="submit"]');
    btn.simulate('submit');
    expect(spyCallback).toHaveBeenCalled();
  });
});
