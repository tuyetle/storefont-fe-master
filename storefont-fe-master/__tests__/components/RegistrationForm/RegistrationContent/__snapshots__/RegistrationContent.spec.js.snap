// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) RegistrationContent should render conrectly 1`] = `
<form
  className="css-1gpz2pu"
  onSubmit={[MockFunction]}
>
  <div
    className="css-sj2822"
  >
     
    register
     
  </div>
  <div
    className="css-193a9w1"
  >
     
    registerDescription
  </div>
  <div
    className="css-3ldr24"
  />
  <div
    className="css-1825lq2"
  />
  <div
    className="form-group"
  >
    <strong
      className={undefined}
    >
      emailOrPhone
    </strong>
    Field
  </div>
  <div
    className="css-1hvtdrp"
  >
    Field
  </div>
  <div>
    <label
      className="css-e5gafv"
    >
      Field
      <span>
        receiveNews
      </span>
    </label>
  </div>
  <input
    className="css-7qje71"
    disabled={true}
    type="submit"
    value="register"
  />
  <div
    className="css-42kidg"
  >
    rulesAndConditionBefore
    <a
      className="css-1ezp9xe"
      href="/"
      onClick={[Function]}
    >
      rulesAndCondition
    </a>
    rulesAndConditionAfter
  </div>
</form>
`;
